### ISSUE TITLE
Fetching last build fails 

### ISSUE THREAD

**Issue by appetrosyan** [orf/cargo-bloat-action#367]
I'm having trouble setting up your action. Particularly, the initial bootstrap, since your action fails at trying to fetch the previous step that doesn't exist. Since I don't see any way in which I can set it up to be the first run, I would imagine that this is one of many usability oversights in your action. 

I don't actually want you to reply to this issue, I want you to ghost it, so that I can convince our team that we don't need your hacky piece of ... generously put.. software, and convince them to write a different action for our specific uses.

**Comment by orf** [comment]
> that we don't need your hacky piece of ... generously put.. software,

How utterly and incredibly rude. Please don't write comments like this.

**Comment by appetrosyan** [comment]
Since you have ghosted every issue that did not contain a direct insult, I wanted to have proof that you actually read them. Now that I know that you do, I hope that postfixing an insult to any and every communication is superfluous.

**Comment by orf** [comment]
Your entitlement is absolutely astounding. Please stop.

**Comment by appetrosyan** [comment]
Thank you for behaving exactly the way I needed you to behave in order to not have to deal with you ever again.

**Comment by orf** [comment]
Just in case you genuinely don't understand why the way you've communicated here isn't unacceptable I'll spell it out for you.

I am absolutely not beholden to reply to any of your messages nor do any work **for** you on this action. It's clearly not been touched recently and there is obviously work that's needed, and **your** pull requests are welcomed.

What's not OK is for you to create an issue after not replying to your previous issue created **one day ago** with the content above. Everyone is going through a tough time with COVID so I'm going to assume the best and put it down to that rather than you being an absolutely incredibly entitled asshole.

I've got my own shit going on and I don't appreciate waking up to a belittling email from a random person I don't know who seems to assume I owe him a response.

In conclusion: happy to review any pull requests, happy for you to fork it, not happy to be insulted by someone trying to use my work for free even if that work isn't complete.

**Comment by appetrosyan** [comment]
It's cool. 

> I am absolutely not beholden to reply to any of your messages nor do any work for you on this action. It's clearly not been touched recently and there is obviously work that's needed, and your pull requests are welcomed.

I can tell that **that** is the case, because of the amount of duplicated issues, and your response (lack thereof) to them.  I understand that it is entitlement to expect you to work on them in your free time, and that you probably didn't have any in a while. It doesn't change the fact that my query would have lingered in limbo. Or that I would have had to have fixed it myself, and had to conform to what you would have deemed acceptable. 

I thought that the worst-case scenario is that you would close the issue and I'd do what I'd have to do anyway, just without having to wait for several weeks, and be beholden to your schedule. 

And I'm quite surprised that you even read them, or noticed the subtle snark. The fact that you did read them, and didn't dignify it with a response... I get it man. I just don't have the time for it. 

> I've got my own shit going on and I don't appreciate waking up to a belittling email from a random person I don't know who seems to assume I owe him a response.

I had no intention to offend you personally. I did clearly state that I only wished to provoke what looks like an emotional response. You closed both issues, which I can use as grounds to start something from a clean slate, rather than work on top of your code. 

Noting personal, but a clear *wontfix* is easier to explain than *maybe hasn't seen the message*. I hate sitting on my arse doing nothing, so apologies if I didn't make it clear in the original message. 

> In conclusion: happy to review any pull requests, happy for you to fork it, 

Thank you for the offer, but I doubt it would be necessary. 

> not happy to be insulted by someone trying to use my work for free even if that work isn't complete.

I don't want to use your work, I want an excuse not to. You've given me plenty, and for that I thank you. 

Also I didn't insult you, I insulted your work... and while I get why that would offend you personally, I would encourage you to disassociate from your open source projects, for you own sanity's sake.

**Comment by orf** [comment]
> or noticed the subtle snark

The point is that it wasn't subtle. It was a very rude message by any standard. Here is what you could have written instead, please try and spot the difference:

> Hey, I see a lot of unanswered issues here and I also can't get it to work. Is this still maintained or is it abandoned? If so could you archive the repository to save me and others some time?

By volume most repositories on GitHub are unmaintained. 

> I don't want to use your work, I want an excuse that I could explain to a non-technical person not to. 

Your terrible sounding work setup, where non-technical people drive choices in libraries and other technical concerns, is none of my business and you should be able to deal with that without involving others.

To keep your sanity on GitHub I suggest improving your ability to spot unmaintained projects by yourself.

**Comment by appetrosyan** [comment]
> Hey, I see a lot of unanswered issues here and I also can't get it to work. Is this still maintained or is it abandoned? If so could you archive the repository to save me and others some time?

Sounds like something I should do in the future. 

On that note, you could have also archived the repo, or had a line somewhere that you hadn't maintained it in a while. Or taken down the good-looking website that prompted this entire discussion in the first place. Or at least added a note saying that you no longer care about it. IDK, something that would have warned people who are not inclined to dig through GitHub repos to  the state of your project. 

In fact, it would be very considerate of you to do it right now. 

> By volume most repositories on GitHub are unmaintained. 

Yes. Most of them are also clearly marked as such. 

> Your terrible sounding work setup, where non-technical people drive choices in libraries and other technical concerns, 

My supervisor is very technical, he just doesn't have the time to dig through the source code. If I said "it looks nasty and barely works" he'd have chalked it up to aesthetics. The fact that the maintainer of the repository is unstable, clearly reads issue comments, but doesn't reply to them, even to say that he won't fix them, is reason enough to build an in-house solution. 

> is none of my business and you should be able to deal with that without involving others.

The fact that you couldn't be bothered to maintain your code shouldn't be mine. 

> To keep your sanity on GitHub I suggest improving your ability to spot unmaintained projects by yourself.

I can tell this project is unmaintained. I can tell that you're a self-important jackass by your choice of the word "entitlement", and the fact that you have

> This makes no sense and is terribly hacky, so sue me. 

in your code. That is also precisely why I knew that adding that insult would prompt a reply, and that's why I wasn't particularly apologetic in my first response. It is also why I'm going to stop explaining myself to you, and enjoy the fact that I can start the work from scratch and not have to deal with you ever again.

**Comment by orf** [comment]
> Or taken down the good-looking website that prompted this entire discussion in the first place. 

There is no website.

> My supervisor is very technical

I'd love to know what he thinks of your communication here. Maybe we could include him in this discussion?

> The fact that the maintainer of the repository is unstable...
> The fact that you couldn't be bothered to maintain your code shouldn't be mine...
> I can tell that you're a self-important jackass...

You are someone who makes contributing to open source software toxic. Please stop.
